src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/SyntaxGeneratorInternalExtensions/SyntaxGeneratorInternal.cs

```diff
@@ -68,5 +68,6 @@ internal SyntaxNode LocalDeclarationStatement(SyntaxToken name, SyntaxNode initi
         internal abstract SyntaxNode InterpolatedStringExpression(SyntaxToken startToken, IEnumerable<SyntaxNode> content, SyntaxToken endToken);
         internal abstract SyntaxNode InterpolationAlignmentClause(SyntaxNode alignment);
         internal abstract SyntaxNode InterpolationFormatClause(string format);
+        internal abstract SyntaxNode TypeParameterList(IEnumerable<string> typeParameterNames);
```

Comment:
uncertani what this is needed for.  There's already SyntaxGEnerator.GenericName, is that not sufficient?

Reply:
This seems easier, just because it is the only part we care about. It's used when displaying the type in the dialog, so as the user types the name will change but the type parameters won't. I guess I could create a SyntaxNode with the new identifier name and then get `SyntaxGenerator.GenericName`, but I didn't see the harm in adding this. 

One point I missed though was why I added this in the first place. ExtractInterface had code doing this in the C# and VB implementations, and I wanted to generalize usage for Extract Class. I need to go and replace those usages now :)